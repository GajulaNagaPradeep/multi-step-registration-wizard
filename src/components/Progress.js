import React from "react";
import { Link, withRouter } from "react-router-dom";

const Progress = ({ location: { pathname } }) => {
  const isStepOne = pathname === "/";
  const isStepTwo = pathname === "/second";
  const isStepThree = pathname === "/third";

  return (
    <React.Fragment>
      <div className="steps">
        <div className={`${isStepOne ? "step active" : "step"}`}>
          <div>1</div>
          <div>
            {isStepTwo || isStepThree ? <Link to="/">Step 1</Link> : "Step 1"}
          </div>
        </div>
        <div className={`${isStepTwo ? "step active" : "step"}`}>
          <div>2</div>
          <div>{isStepThree ? <Link to="/second">Step 2</Link> : "Step 2"}</div>
        </div>
        <div className={`${pathname === "/third" ? "step active" : "step"}`}>
          <div>3</div>
          <div>Step 3</div>
        </div>
      </div>
    </React.Fragment>
  );
};

export default withRouter(Progress);
