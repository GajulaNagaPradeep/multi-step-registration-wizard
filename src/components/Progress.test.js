import React from "react";
import { shallow } from "enzyme";
import Progress from "./Progress";

const props = { location: { pathname: "/" } };

let wrapped = shallow(<Progress {...props} />);
describe("Progress", () => {
  it("should be rendered correctly", () => {
    expect(wrapped).toMatchSnapshot();
  });
});
