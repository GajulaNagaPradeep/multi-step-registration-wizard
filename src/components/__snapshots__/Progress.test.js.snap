// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress should be rendered correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
